src/storage.rs

```diff
@@ -63,6 +66,53 @@ pub struct UpdateEntryJson {
     pub public_key: String,
 }
 
+fn decode_signed_message(signed_message: &String) -> Result<Vec<u8>, DeimosError> {
+    let signed_message_bytes = general_purpose::STANDARD
+        .decode(&signed_message)
+        .map_err(|_| DeimosError::General(GeneralError::DecodingError("failed to decode signed message".to_string())))?;
+
+    // check if the signed message is (at least) 64 bytes long
+    if signed_message_bytes.len() < 64 {
+        return Err(DeimosError::General(GeneralError::ParsingError(
+            "signed message is too short".to_string(),
+        )));
+    } else {
+        Ok(signed_message_bytes)
+    }
+}
+
+impl Signable for UpdateEntryJson {
+    fn get_signature(&self) -> Result<Signature, DeimosError> {
+        let signed_message_bytes = decode_signed_message(&self.signed_message)?;
+
+        // extract the first 64 bytes from the signed message which are the signature
+        let signature_bytes: &[u8; 64] = match signed_message_bytes.get(..64) {
+            Some(array_section) => match array_section.try_into() {
+                Ok(array) => array,
+                Err(_) => Err(DeimosError::General(GeneralError::ParsingError(
+                    "failed to convert signed message to array".to_string(),
+                )))?,
+            },
+            None => Err(DeimosError::General(GeneralError::ParsingError(
+                "failed to extract signature from signed message".to_string(),
+            )))?,
+        };
+
+        Ok(Signature::from_bytes(signature_bytes))        
+    }
+
+    fn get_content_to_sign(&self) -> Result<String, DeimosError> {
+        let signed_message_bytes = decode_signed_message(&self.signed_message)?;
+        let message_bytes = &signed_message_bytes[64..];
+        Ok(String::from_utf8_lossy(message_bytes).to_string()) // or from_utf8_unchecked?
```

Comment:
resolve this question first

Reply:
true, I would use `from_utf8_lossy()`. If the characters are valid in the sense of utf8 (as desired), a byte slice is converted into a string. all invalid uft8 sequences are replaced by a substitute character, but this way the behavior is still always understandable and not "unsafe"